python/peacock/ExodusViewer/plugins/VariablePlugin.py

```diff
@@ -315,6 +315,17 @@ def __setVariableList(self, reader):
         Helper for updating the variable list while maintaining the current selection
         """
         variables = self._reader.getVariableInformation(var_types=[self._reader.NODAL, self._reader.ELEMENTAL])
+
+        # We want to avoid clearing the list and rebuilding it as that
+        # can cause confusion if somebody is in the middle of a selection
+        new_vars = {v.name for v in variables.values()}
+        cur_vars = {self.VariableList.itemText(i) for i in range(self.VariableList.count())}
+        if new_vars == cur_vars:
+            for vinfo in variables.itervalues():
+                idx = self.VariableList.findText(vinfo.name) # should always return a valid index
+                self.VariableList.setItemData(idx, vinfo)
+            return
+
         current = self.VariableList.currentText()
```

Comment:
Can you make this the "else" portion of the if statement, just to avoid the return in the middle of the function. I didn't see it initially at was a bit confused why the function continued on a re-built the list any way.